### DISCUSSION TITLE
CYD as meshtastic beeper via Lora integration

### DISCUSSION THREAD

**Discussion by BrazenCodebane** [witnessmenow/ESP32-Cheap-Yellow-Display#175] [Ideas]
I'm a programmer not an EE anyone willing to guide a newbie?

**Comment by BrazenCodebane** [comment]
Looking into the reyax rylr998 module which has a UART connection. Pins: VDD,RST,TX,RX,GND.  Seems to operate VDD on 3.3v. 
Would the IO on the CYD function as UART or is the 4 pin connector better option?

Forgive my newbieness.

**Comment by aly-fly** [comment]
ESP32 (processing unit on CYD) has 3 UART ports. One is hard wired to the USB converter chip, used for programming and serial monitor (simple debug). In code referred to as `Serial`. The second and third UART are free to use for any hardware extension and is used in code by references `Serial1` and `Serial2`. Which pins are best to use for both ports is up to you to find out. Read through the pinout docs. Fortunately, most of functions can be remapped to other IO pins.

**Reply by BrazenCodebane** [reply]
Thats awesome.. that alone makes it worth investing in to test the project.Thanks for the guidance.

**Reply by BrazenCodebane** [reply]
looking at the documentation seems like the CN1 bus would be the best candidate pinout mentions good for I2C, which seems to be another serial protocol...

**Reply by aly-fly** [reply]
https://electropeak.com/learn/full-guide-to-esp32-pinout-reference-what-gpio-pins-should-we-use/
Looking at the docs, there are 3 UART ports. I have updated my first comment.

U0 is the debug port.
U1 is is connected to SP hardware pins 17 and 18 (gpio 9 and 10). 
U2 is is connected to SP hardware pins 27 and 28 (gpio 16 and 17). 

Best is to use these pins. If not accessible, you might be able to remap them, but this has to be verified, how flexible it is.

**Comment by BrazenCodebane** [comment]
Heres what Ive gotten together on the project so far.
TFT_eTouch library needs to be added to platformIO (kindof a roadblock since new) 
and the specific board needs to be "supported" on platformIO to be able to pull in all necessary libraries for the end project


Anyone have any skills on platformIO who can help me out?


#ifndef _VARIANT_CYD24_ESP32_TFT
#define _VARIANT_CYD24_ESP32_TFT

#include <TFT_eTouch.h>
// touch section
#define ETOUCH_MOSI 32
#define ETOUCH_MISO 39
#define ETOUCH_SCK 25
#define ETOUCH_CS 33
#define ETOUCH_IRQ 0xff // not used in this sketch

SPIClass hSPI(HSPI); //define SPI port to be used by TFT_eTouch

TFT_eSPI tft = TFT_eSPI();
TFT_eTouch<TFT_eSPI> touch(tft, ETOUCH_CS, ETOUCH_IRQ, hSPI);

//display section

#define TFT_DC 2
#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCK 14
#define TFT_CS 15
#define TFT_BL 21


//LoRa Section
#define LORA_RX     //TODO: pinout
#define LORA_TX     //TODO: pinout


// hardware section

#define BOOT 0 

#endif _VARIANT_CYD24_ESP32_TFT

**Comment by BrazenCodebane** [comment]
UPDATE: going to attempt to support the specific board on PlatformIO. but have yet to purchase the board. Can anyone give me specific details such as the microcontroller model, clock speed, flash size,and pin mapping for a board definition file on the 2.4 inch type c variant
of the CYD?

**Comment by BrazenCodebane** [comment]
anyone have a quick fix for this issue on custom firmware...
0 I2C device found at address 0x15
0 Device found at address 0x15 was not able to be enumerated


model:  ESP32-2432S024C
no screen activation due to failed enumeration

**Comment by gringobenuk** [comment]
Also interested to integrate LoRa with the CYD. I have the boards and am looking for pointers from people who have done it.

**Reply by justicestorm** [reply]
Me too, and a card puter

**Comment by teletonn** [comment]
any success in CYD + Lora? I'm using Lilygo T3-S3, but its have a very little display and only one user button.. It would be greate tu use this: https://github.com/meshtastic/device-ui on it. But there is only a photo with CYD and I'm too newb to find out how to do that. Maybe some easy way exist?
